File: TP3/src/main/scala/counter/Counter.scala
package counter

import java.util.Observable

case class Counter(var name: String, var _count: Int) extends Observable {

  def count: Int = _count

  def increment() {
    System.out.println("increment");
    _count += 1
    setChanged()
    notifyObservers(count)
  }

  def decrement() {
    System.out.println("decrement");
    _count -= 1
    setChanged()
    notifyObservers(count)
  }
}

File: TP3/src/main/java/counter/Counter.java
package counter;
import java.util.Observable;

public class Counter extends Observable {
	public String name;
	private int count;

	public Counter(String name, int count) {
		this.name = name;
		this.count = count;
	}
	int getCount() {
		return count;
	}
	void increment() {
//		System.out.println("increment");
		count++;
		setChanged();
		notifyObservers(count);
	}
	void decrement() {
//		System.out.println("decrement");
		count--;
		setChanged();
		notifyObservers(count);
	}
}
